server/use_cases/add_service_commitments.py

```diff
@@ -0,0 +1,55 @@
+"""
+Module for handling service commitments.
+Provides functions to create and retrieve service commitments for users.
+"""
+
+import uuid
+
+def add_service_commitments(repo, user_email, shifts):
+    """
+    Creates service commitments for the given user and shifts.
+
+    Args:
+        repo: A repository object that provides 
+        an insert_service_commitments method.
+        user_email (str): The user's email.
+        shifts (list): A list of shift dictionaries.
+
+    Returns:
+        list: A list of dictionaries indicating
+        the success and service commitment IDs.
+    """
+    commitments = []
+    for shift in shifts:
+        if "service_shift_id" not in shift:
+            continue
+
+        commitment = {
+            "service_commitment_id": str(uuid.uuid4()),
+            "service_shift_id": shift["service_shift_id"],
+            "user_email": user_email
+        }
```

Comment:
We don't want to generate the service commitment id that our application generates into the repo. We want the repo to assign the service commitment id.